File: js/advanced/4b_two_sum_sorted_array.js
// Find two nos. in the sorted array which sum up to int target (& return their 1-based indeces);

var twoSum = function(numbers, target) {
    let l = 0, r = numbers.length - 1
    while(l<r){
        if(numbers[l] + numbers[r] == target) return [l+1, r+1]
        else if(numbers[l] + numbers[r] < target) l++
        else r--
    }
};

File: c++/advanced/4b_two_sum_sorted_array.cpp
// Find two nos. in the sorted array which sum up to int target (& return their 1-based indeces);

vector<int> twoSum(vector<int>& numbers, int target) {
        int l=0, r=numbers.size()-1;
        while(l<r){
            if(numbers[l] + numbers[r] == target) return {l+1, r+1};
            else if(numbers[l] + numbers[r] < target) l++;
            else r--;
        }
        return {0,0};
    }